hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -886,11 +887,15 @@ public int getMemberListJoinVersion() {
             if (!isJoined()) {
                 throw new IllegalStateException("Member list join version is not available when not joined");
             } else if (getClusterVersion().isLessThan(V3_10)) {
-                String msg = "Member list join version is not available with the cluster version less than 3.10";
+                String msg = "Member list join version is not available with a cluster version less than 3.10";
                 throw new IllegalStateException(msg);
             }
 
-            return localMember.getMemberListJoinVersion();
+            int joinVersion = localMember.getMemberListJoinVersion();
+            if (joinVersion == NA_MEMBER_LIST_JOIN_VERSION) {
+                throw new IllegalStateException("Member list join version is not yet available");
```

Comment:
we don't need this. if all previous checks are passed, then it means the member list join version is available...

Reply:
No, it's the case when cluster turns from 3.9 to 3.10 and the member did not yet learn it's join version by an async call from master. It's the case where we previously returned `NA_MEMBER_LIST_JOIN_VERSION` to the caller; i changed it to throwing to have all checks in one place. I added a comment.

Reply:
ok, you are right